src/deprecated/openai_assistant_deprecated.rs

```diff
@@ -0,0 +1,799 @@
+use anyhow::{anyhow, Context, Result};
+use log::{error, info};
+use reqwest::{
+    header::{self, HeaderMap, HeaderValue},
+    multipart, Client,
+};
+use schemars::{schema_for, JsonSchema};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::path::Path;
+use std::time::Duration;
+use tokio::time;
+use tokio::time::timeout;
+
+use super::OpenAIModels;
+use crate::constants::{OPENAI_API_URL, OPENAI_ASSISTANT_INSTRUCTIONS};
+use crate::domain::{
+    OpenAIAssistantResp, OpenAIMessageListResp, OpenAIMessageResp, OpenAIRunResp, OpenAIThreadResp,
+};
+use crate::enums::{OpenAIAssistantRole, OpenAIRunStatus};
+use crate::utils::sanitize_json_response;
+
+/// This is a DEPRECATED implementation of OpenAI Assistants API that will not be maintained going forward (after May 2024).
+/// For current implementation, including support for Assistants API v2 and GPT-4o, refer to `assistants` module.
+///
+/// [OpenAI Docs](https://platform.openai.com/docs/assistants/overview)
+///
+/// The Assistants API allows you to build AI assistants within your own applications.
+/// An Assistant has instructions and can leverage models, tools, and knowledge to respond to user queries.
+/// The Assistants API currently supports three types of tools: Code Interpreter, Retrieval, and Function calling.
+/// In the future, we plan to release more OpenAI-built tools, and allow you to provide
+/// your own tools on our platform.
+#[deprecated(
+    since = "0.6.1",
+    note = "This struct is deprecated. Please use the `assistants::OpenAIAssistant` struct for latest functionality including Assistants API v2."
+)]
```

Comment:
@darioalessandro,

I added this script to all deprecated public structs. This will result in a warning if someone is trying to use any of the struct and point them to the correct alternative.

If you run the examples marked as 'deprecated_' you will get those warnings too. There should be no warnings for the standard examples. 

Now, those 'deprecated_' examples make the CI fail. Any thoughts on what should I do with them? Suppressing deprecation warnings is an option but is that the right call? Or I can just remove the examples with deprecated structs?